src/core/sync/mutex.d

```diff
@@ -55,62 +57,92 @@ class Mutex :
     /**
      * Initializes a mutex object.
      *
-     * Throws:
-     *  SyncError on error.
      */
-    this() nothrow @trusted
+    this() @trusted nothrow @nogc
+    {
+        this(true);
+    }
+
+    /// ditto
+    this() shared @trusted nothrow @nogc
```

Comment:
This indeed looks like a compiler bug. If the base class has multiple default constructors, the compiler should synthesize multiple default constructors in the derived class. I.e. the base class and the derived class should have equivalent default constructor overload sets.

Do you think that the front-end can be fixed in a relatively short-term period? If not, probably the best course of action would be to revert this PR, though I prefer not to.

Hopefully the project tester can prevent such issues from coming up next time. I wonder why it didn't run for the whole month my PR was up for review.